src/extension.ts

```diff
@@ -124,12 +127,18 @@ export async function activate(context: vscode.ExtensionContext): Promise<vscode
 			});
 		}
 	});
-	context.subscriptions.push(runOnActiveFile, runOnSelected, runDfaOnSelectedMethod, removeDiagnosticsOnActiveFile, removeDiagnosticsOnSelectedFile);
+	context.subscriptions.push(runOnActiveFile, runOnSelected, runDfaOnSelectedMethod, removeDiagnosticsOnActiveFile, removeDiagnosticsOnSelectedFile, removeSingleDiagnostic);
 	TelemetryService.sendExtensionActivationEvent(extensionHrStart);
 	outputChannel.appendLine(`Extension sfdx-code-analyzer-vscode activated.`);
 	return Promise.resolve(context);
 }
 
+export function _removeSingleDiagnostic(uri: vscode.Uri, diagnosticToRemove: vscode.Diagnostic, diagnosticCollection: vscode.DiagnosticCollection) {
+	const currentDiagnostics = diagnosticCollection.get(uri) || [];
+	const updatedDiagnostics = currentDiagnostics.filter(diagnostic => diagnostic != diagnosticToRemove);
```

Comment:
If you are just removing a single diagnostic, then I would recommend not using filter (because it doesn't stop after it finds the first one to remove). I'm guessing typescript has a findIndex or a remove method or something that you can you use instead maybe. Just a thought - not blocking - feel free to still submit this if you feel it's most elegant.